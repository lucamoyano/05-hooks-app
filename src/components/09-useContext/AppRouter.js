import React from 'react';
import {
    BrowserRouter as Router,
    Switch,
    Route,
    Redirect
  } from "react-router-dom";

import { NavBar } from './NavBar';

import { AboutScreen } from './AboutScreen';
import { HomeScreen } from './HomeScreen';
import { LoginScreen } from './LoginScreen';

export const AppRouter = () => {
    return (
        <Router>
            <div>

                <NavBar />

                
                {/* Propiedad 'exact' para que renderice el componente si el path es exacto */}
                    <Switch>
                        <Route exact path="/" component={ HomeScreen } />
                        <Route exact path="/about" component={ AboutScreen } />
                        <Route exact path="/login" component={ LoginScreen } />
                        
                        {/* En caso que el path no exista componente por defecto*/}
                        {/* <Route component={ HomeScreen } /> */}

                        {/* En caso que el path no exista redirecciona*/}
                        <Redirect to="/" />
                    </Switch>
                
                
            </div>
        </Router>
    )
}
