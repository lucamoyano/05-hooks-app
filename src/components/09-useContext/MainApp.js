import React from 'react'

export const MainApp = () => {
    return (
        <div>
            <h1>MainScreen</h1>
            <hr />
        </div>
    )
}
