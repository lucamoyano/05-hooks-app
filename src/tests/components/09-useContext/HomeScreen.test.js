import { mount} from 'enzyme';

import { HomeScreen } from '../../../components/09-useContext/HomeScreen';
import { UserContext } from "../../../components/09-useContext/UserContext";

describe('Pruebas en <HomeScreen />', ()=> {
    
    const user = {
        id: 1234,
        name: 'Lucas',
        email: 'lucanmoyano@gmail.com'
    }

    const wrapper = mount( 
        <UserContext.Provider
            value={{
                user
            }}>
            <HomeScreen />
        </UserContext.Provider>
    );

    test('debe mostrarse correctamente', () => {
        expect( wrapper ).toMatchSnapshot();
    });
    
})