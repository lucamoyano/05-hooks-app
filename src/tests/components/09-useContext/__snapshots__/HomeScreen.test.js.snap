// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <HomeScreen /> debe mostrarse correctamente 1`] = `
<div>
  <h1>
    HomeScreen
  </h1>
  <hr />
  <pre>
    {
   "id": 1234,
   "name": "Lucas",
   "email": "lucanmoyano@gmail.com"
}
  </pre>
</div>
`;
